src/Service/AssistantService.php

```diff
@@ -0,0 +1,331 @@
+        if (array_key_exists($step, $steps) && array_key_exists('newTitle', $steps[$step])) {
+            return $this->translator->trans($steps[$step]['newTitle'], domain: 'assistant');
+        }
+        return null;
+    }
+
+    public function getElementTypeForStep(int $step): ?string {
+        $steps =  AssistantService::$STEPS;
+        if (array_key_exists($step, $steps) && array_key_exists('type', $steps[$step])) {
+            return $steps[$step]['type'];
+        }
+        return null;
+    }
+
+    public function getContactForStep(int $step): ?Kontakte {
+        $steps =  AssistantService::$STEPS;
+        if (array_key_exists($step, $steps) && array_key_exists('contact', $steps[$step])) {
+            $contactStep = $steps[$step]['contact'];
+            $contactId = $this->getPropertyForStep($contactStep);
+            return $contactId ? $this->contactRepository->find($contactId): null;
+        }
+        return null;
+    }
+
+    public function getProcedureForStep(int $step): ?VVT {
+        $steps =  AssistantService::$STEPS;
+        if (array_key_exists($step, $steps) && array_key_exists('vvt', $steps[$step])) {
+            $procedureStep = $steps[$step]['vvt'];
+            $procedureId = $this->getPropertyForStep($procedureStep);
+            return $procedureId ? $this->vvtRepository->find($procedureId): null;
+        }
+        return null;
+    }
+
+    public function getSoftwareForStep(int $step): ?Software {
+        $steps =  AssistantService::$STEPS;
+        if (array_key_exists($step, $steps) && array_key_exists('software', $steps[$step])) {
+            $softwareStep = $steps[$step]['software'];
+            $softwareId = $this->getPropertyForStep($softwareStep);
+            return $softwareId ? $this->softwareRepository->find($softwareId): null;
+        }
+        return null;
+    }
+
+    public function getSkipForStep(int $step): ?bool {
+        $steps =  AssistantService::$STEPS;
+        if (array_key_exists($step, $steps) && array_key_exists('skip', $steps[$step])) {
+            return $steps[$step]['skip'];
+        }
+        return false;
+    }
+
+    public function getStepCount(): int {
+        return count(AssistantService::$STEPS);
+    }
+
+    public function setStep(int $step) {
+        $session = $this->requestStack->getSession();
+        $session->set('step', $step);
+    }
+
+    public function getStep(): int {
+        $session = $this->requestStack->getSession();
+        $id = $session->get('step');
+        return $id ? : 0;
+    }
+
+    public function clear() {
+        $session = $this->requestStack->getSession();
+        $session->set('step', 0);
+
+        foreach (AssistantService::$STEPS as $step) {
+            $session->remove($step['key']);
+        }
+    }
+
+    public function createElementForStep(int $step, User $user, Team $team)
+    {
+        $item = null;
+
+        switch ($this->getElementTypeForStep($step)) {
```

Comment:
remove the returns in the cases since you can use break and return $item at the end of the method